File: src/Framework/DrawContext.fs
namespace Animations

open System
open SkiaSharp

type public DrawContext = 
    struct
        val Properties: AnimationProperties
        val FrameNumber: int
        val Time: TimeSpan
        val Canvas: SKCanvas

        new (properties: AnimationProperties, frameNumber: int, time: TimeSpan, canvas: SKCanvas) = 
            { Properties = properties; FrameNumber = frameNumber; Time = time; Canvas = canvas }
    end

File: Animations.Framework/DrawContext.cs
using System;
using SkiaSharp;

namespace Animations.Framework
{
    public interface IDrawContext
    {
        AnimationProperties AnimationProperties { get; }
        int FrameNumber { get; }
        TimeSpan Time { get; }
        SKCanvas Canvas { get; }
    }

    internal class DrawContext : IDrawContext
    {
        public AnimationProperties AnimationProperties { get; set; }
        public int FrameNumber { get; set; }
        public TimeSpan Time { get; set; }
        public SKCanvas Canvas { get; set; }
    }
}